fplll/pruner/pruner.h

```diff
@@ -254,8 +316,42 @@ template <class FT> class Pruner
       symmetry_factor = 1;
     }
     min_pruning_coefficients.resize(d);
+    btmp.resize(d);
+    bftmp.resize(n);
     fill(min_pruning_coefficients.begin(), min_pruning_coefficients.end(), 0.);
     set_tabulated_consts();
+
+    // optimize overall cost, use
+    if (flags & PRUNER_SINGLE)
+    {
+      opt_single = true;
+    }
+
+    // need to fix target if possible
+    if (metric == PRUNER_METRIC_PROBABILITY_OF_SHORTEST)
+    {
+      // check if the target is reasonable; otherwise set it to 0.99.
+      // note the target may affect the cost in target_function().
+      if (this->target > 1.0 || this->target < 0.0)
+      {
+        this->target = 0.99;
+      }
+    }
+    else if (metric == PRUNER_METRIC_EXPECTED_SOLUTIONS)
+    {
+      // check if the target is reasonable; otherwise set it to 0.99.
+      // note the target is allowed to be larger than 1 in this case.
+      // Also the target may affect the cost in target_function().
+      if (this->target < 0.0)
+      {
+        this->target = 0.99;
```

Comment:
Should we not throw an error here? Otherwise, I'd expect target = 1.0 to be a good choice?

Reply:
yes, it throws an exception now.